### PR TITLE
Enable transaction dispute details feature for all merchants (remove feature flag)

### LINKED ISSUE #7289: Transaction Details → Dispute Details – Remove feature flag
Part of #6883

We need to remove the feature flag `_wcpay_feature_dispute_on_transaction_page` to allow merchants to access dispute details on the Transaction Details screen.

- Create changelog entry
- [Update any affected critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows)
- [Create release testing instructions](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions)

### PR DESCRIPTION
**Pull request by Jinksi**
Fixes #7289

#### Changes proposed in this Pull Request

This PR enables the transaction dispute details feature for all merchants by removing the feature flag `_wcpay_feature_dispute_on_transaction_page`.

#### TODO:
- [x] Create changelog entry
	`Display dispute information, recommended resolution steps, and actions directly on the transaction details screen to help merchants with dispute resolution.`

Post-merge:
- [x] [Update any affected critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows)
- [x] [Create release testing instructions](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions): https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-6.6.0#enable-transaction-dispute-details-feature-for-all-merchants-pr-7365

**Before**
image

**After**
image

#### Testing instructions

* Ensure you've disabled the feature flag `_wcpay_feature_dispute_on_transaction_page` if previously enabled
	* `delete_option( '_wcpay_feature_dispute_on_transaction_page' )`
* Visit a dispute details screen and ensure it redirects to the transaction details screen
* Ensure the dispute details section appears on the transaction details screen for
	* Dispute awaiting a response
	* Dispute under review
	* Dispute lost
	* Dispute won
	* Inquiry awaiting a response
	* Inquiry under review
	* Inquiry closed

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [x] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions-for-WC-Payments-6.6.0#enable-transaction-dispute-details-feature-for-all-merchants-pr-7365
- [x] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [x] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7365` or branch name `add/7289-remove-transaction-dispute-details-feature-flag` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/7289-remove-transaction-dispute-details-feature-flag) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: bebbecd64992feeaa111e64f6305bbf0e92f6ac5
- Build time: 2023-10-05 00:09:18 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/payment-details/summary/test/index.test.tsx [hunk 1]
```diff
@@ -461,390 +358,461 @@ describe( 'PaymentDetailsSummary', () => {
...
+	} );
 
-			// Don't render the staged evidence message
-			expect(
-				screen.queryByText(
-					/You initiated a challenge to this dispute/,
-					{ ignore: '.a11y-speak-region' }
-				)
-			).toBeNull();
+	test( 'renders the information of a dispute-reversal charge', () => {
```
Comment (Jinksi) [inline client/payment-details/summary/test/index.test.tsx:465]: This is not a new test, simply moved down in this file to group dispute-related tests.

File: changelog/add-7289-remove-transaction-dispute-details-feature-flag [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: major
+Type: add
+
+Display dispute information, recommended resolution steps, and actions directly on the transaction details screen to help merchants with dispute resolution.
```
Comment (haszari) [inline changelog/add-7289-remove-transaction-dispute-details-feature-flag:4]: Nice changelog!

File: client/index.js [hunk 3]
```diff
@@ -152,14 +151,8 @@ addFilter(
...
 
-		// If disputes on transaction page feature is enabled, set up a soft
-		// redirect component; otherwise register the (legacy) dispute details page.
-		const isDisputeOnTransactionPageEnabled =
-			window.wcpaySettings.featureFlags.isDisputeOnTransactionPageEnabled;
 		pages.push( {
-			container: isDisputeOnTransactionPageEnabled
-				? RedirectToTransactionDetails
-				: LegacyDisputeDetailsPage,
+			container: RedirectToTransactionDetails,
```
Comment (haszari) [inline client/index.js:155]: We can also remove `LegacyDisputeDetailsPage` completely.

**Review by haszari — APPROVED** [review]
Did a quick test of various dispute flows after deleting feature flag.

```
% docker-compose exec wordpress wp --allow-root option delete _wcpay_feature_dispute_on_transaction_page
Success: Deleted '_wcpay_feature_dispute_on_transaction_page' option.
% docker-compose exec wordpress wp --allow-root option get _wcpay_feature_dispute_on_transaction_page  
Error: Could not get '_wcpay_feature_dispute_on_transaction_page' option. Does it exist?
%
```

- View payments overview, click `You have many disputes` notice, land on disputes list.
- Click a dispute, land on relevant transaction with inline dispute info (via redirect spinner). I tested inquiries and disputes with a range of dates.
- Inquiry:
  - Click `Submit evidence`, land on dispute response form.
  - Click `Issue refund`, shows a modal, can click through to relevant order.
- Dispute: 
  - Click `Challenge dispute`, land on dispute response form.
  - Click `Accept dispute`, shows a modal.
    - Click `Accept dispute`, eventually shows transaction detail with resolved dispute state.

I didn't search the code for the feature flag – assuming you've done that :) 

One blocker (IMO):

- Delete `LegacyDisputeDetailsPage`, no longer needed.

Some minor design notes, might want to log follow up maintenance issues:

- Extra line at bottom of respond steps, e.g. between `Submit evidence or issue a refund` and action buttons. This may be as designed, feels disconnected/cluttered. I wonder if we need lines between steps at all.
- `Accept dispute` flow is a little disconnected, might benefit from a transition or spinner after the modal.

@nikkivias FYI – design feedback / notes for your consideration

File: client/index.js (hunk 3 above)
Comment (Jinksi) [reply client/index.js:155]: I believe this is being handled in #7363 and is OK to merge after 6.6.

**Comment by Jinksi** [comment]
> Accept dispute flow is a little disconnected, might benefit from a transition or spinner after the modal.

Agreed! This came up in our UX review. I will log an issue for this if it hasn't been created already.

**Comment by haszari** [comment]
> Agreed! This came up in our UX review. I will log an issue for this if it hasn't already.

Probably ok to leave that with product / design and let it come up as a project / maintenance via normal process. GitHub is best for things we know we want to prioritise.

**Comment by Jinksi** [comment]
Note this PR has resulted in some failing e2e tests. These tests are being skipped in #7088.

**Pull request merged**